src/libopenrave/planner.cpp

```diff
@@ -180,6 +174,18 @@ PlannerStatus& PlannerStatus::SetPlannerParameters(PlannerParametersConstPtr par
     return *this;
 }
 
+// This assumes that collisionReport plink1 and plink2 fields are populated! Current implementation does not check if vLinkColliding is populated (when is it populated)?
+void PlannerStatus::UpdatePlannerStatusInfo(CollisionReport& collisionReport)
+{      
+    if (!!collisionReport.plink1 && !!collisionReport.plink2) {
+        std::pair<KinBody::LinkConstPtr,KinBody::LinkConstPtr> collisionPair(collisionReport.plink1, collisionReport.plink2);
+        if (mCollidingLinksCount.find(collisionPair) == mCollidingLinksCount.end()) {
```

Comment:
Efficiency can be improved here. You did a `find()` operation, you might as well use the resulting iterator to set the value. Doing another `mCollidingLinksCount[collisionPair]` to increment means you are finding the same iterator again.